src/gazebo_geotagged_images_plugin.cpp

```diff
@@ -578,12 +590,98 @@ void GeotaggedImagesPlugin::_handle_request_camera_settings(const mavlink_messag
+        MAVLINK_COMM_1,
+        &msg,
+        static_cast<uint8_t>(sid),                              // Stream ID
+        VIDEO_STREAM_STATUS_FLAGS_RUNNING,                      // Flags (It's always running)
+        30,                                                     // Frame rate
+        _width,                                                  // Horizontal resolution
+        _height,                                                // Vertical resolution
+        2048,                                                   // Bit rate (made up)
+        0,                                                      // Rotation (none)
+        90                                                      // FOV (made up)
+    );
+    _send_mavlink_message(&msg, srcaddr);
+}
+
+void GeotaggedImagesPlugin::_handle_request_video_stream_information(const mavlink_message_t *pMsg, struct sockaddr* srcaddr)
+{
+    gzdbg << "Send videostream information" << endl;
+    char name[MAVLINK_MSG_VIDEO_STREAM_INFORMATION_FIELD_NAME_LEN] = {};
+    snprintf(name, MAVLINK_MSG_VIDEO_STREAM_INFORMATION_FIELD_NAME_LEN, "Visual Spectrum");
+    mavlink_command_long_t cmd;
+    mavlink_msg_command_long_decode(pMsg, &cmd);
+    //-- Should we execute the command
+    if ((int)cmd.param1 != 1)
+    {
+        gzwarn << "Request ignored" << endl;
+        return;
+    }
+
+    // ACK command received and accepted
+    _send_cmd_ack(pMsg->sysid, pMsg->compid, MAV_CMD_REQUEST_VIDEO_STREAM_INFORMATION, MAV_RESULT_ACCEPTED, srcaddr);
+    std::string uri = "5600";
+
+    mavlink_message_t msg;
+    mavlink_msg_video_stream_information_pack_chan(
+        1,
+        MAV_COMP_ID_CAMERA,                         // Component ID
+        MAVLINK_COMM_1,
+        &msg,
+        1,                                          // Stream ID
+        1,                                          // Stream count
+        VIDEO_STREAM_TYPE_RTPUDP,                   // Stream type
+        VIDEO_STREAM_STATUS_FLAGS_RUNNING,          // Flags (It's always running)
+        30,                                         // Frame rate
+        _width,                                     // Horizontal resolution
+        _height,                                    // Vertical resolution
+        2048,                                       // Bit rate
+        0,                                          // Rotation (none)
+        90,                                         // FOV (made up)
+        name,
+        uri.c_str()
+    );
+    _send_mavlink_message(&msg, srcaddr);
+}
+
+void GeotaggedImagesPlugin::_handle_set_camera_mode(const mavlink_message_t *pMsg, struct sockaddr* srcaddr)
+{
+    mavlink_command_long_t cmd;
+    mavlink_msg_command_long_decode(pMsg, &cmd);
+    _send_cmd_ack(pMsg->sysid, pMsg->compid,
+                  MAV_CMD_SET_CAMERA_MODE, MAV_RESULT_ACCEPTED, srcaddr);
+    if (_mode == CAMERA_MODE_VIDEO) {
+        _mode = CAMERA_MODE_IMAGE;
```

Comment:
Is this right? Shouldn't it be:

```suggestion
        _mode = CAMERA_MODE_VIDEO;
```

of course bellow would be
```c++
        _mode = CAMERA_MODE_IMAGE;
```